lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,1089 @@
+		desc:SetAutoStretchVertical(true)
+		desc:SetWrap(true)
+		desc:Dock(TOP)
+		if message then
+			local descGivenDisclaimer = vgui.Create("DLabel", self)
+			descGivenDisclaimer:SetText("The following reason was provided:")
+			descGivenDisclaimer:SetAutoStretchVertical(true)
+			descGivenDisclaimer:DockMargin(0, 5, 0, 0)
+			descGivenDisclaimer:Dock(TOP)
+			local descGiven = vgui.Create("DTextEntry", self)
+			descGiven:SetHeight(60)
+			descGiven:SetText(message)
+			descGiven:SetEnabled(false)
+			descGiven:DockMargin(0, 5, 0, 0)
+			descGiven:Dock(TOP)
+		end
+		local descExpires = vgui.Create("DLabel", self)
+		descExpires:SetText("This request expires in XXX seconds.")
+		descExpires:SetWrap(true)
+		descExpires:DockMargin(0, 5, 0, 0)
+		descExpires:Dock(TOP)
+		local blockRequests = vgui.Create("DCheckBoxLabel", self)
+		blockRequests:SetText("Block future money requests from this user")
+		blockRequests:SetWrap(true)
+		blockRequests:DockMargin(0, 5, 0, 0)
+		blockRequests:Dock(TOP)
+		local buttons = vgui.Create('Panel', self)
+		buttons:DockMargin(0, 5, 0, 0)
+		buttons:Dock(BOTTOM)
+		local btnDecline = vgui.Create("StarfallButton", buttons)
+		btnDecline:Dock(LEFT)
+		btnDecline:SetText("Decline")
+		btnDecline:SetAutoSize(false)
+		btnDecline:SetWidth(select(2, self:GetSize())/2)
+		function btnDecline.DoClick()
+			RunConsoleCommand("sf_moneyrequest", 0, "decline", index, expiry)
+			self:Close()
+		end
+		local btnAccept = vgui.Create("StarfallButton", buttons)
+		btnAccept:Dock(RIGHT)
+		btnAccept:SetText("Accept (Wait XXX seconds)")
+		btnAccept:SetAutoSize(false)
+		btnAccept:SetWidth(select(2, self:GetSize())/2)
+		function btnAccept.DoClick()
+			RunConsoleCommand("sf_moneyrequest", 0, "accept", index, expiry)
+			self:Close()
+		end
+		local receiverSteamID = receiver:SteamID() -- In case they disconnect between now and the window closing
+		function self:OnClose()
+			if blockRequests:GetChecked() then
+				blocked[receiverSteamID] = true
+			end
+		end
+		local me = LocalPlayer()
+		local function updateAcceptEnabled(bool)
+			local timeUntilSafe = 2-(SysTime()-startTime)
+			if timeUntilSafe >= 0 then
+				btnAccept:SetEnabled(false)
+				btnAccept:SetText(string.format("Accept (Wait %s)", string.NiceTime(math.ceil(timeUntilSafe))))
+			else
+				btnAccept:SetText("Accept")
+				if not me:canAfford(amount) then
+					btnAccept:SetEnabled(false)
+				else
+					if bool == nil then
+						bool = blockRequests:GetChecked()
+					end
+					btnAccept:SetEnabled(not bool)
+				end
+			end
+		end
+		blockRequests.OnChange = updateAcceptEnabled
+		updateAcceptEnabled()
+		function self:Think()
+			updateAcceptEnabled()
+			local remaining = expiry-CurTime()
+			descExpires:SetText(string.format("This request expires in %s.", string.NiceTime(remaining)))
+			if remaining <= 0 then
+				self:Close()
+			end
+		end
+		self:Open()
+	end
+	vgui.Register("StarfallMoneyRequestFrame", PANEL, "StarfallFrame")
```

Comment:
Instead of registering, you could put all this in a `local function createMoneyRequest()` that creates the StarfallFrame and does all the stuff in 'init'